lib/Chunk.js

```diff
@@ -34,50 +75,104 @@ const getModulesIdent = set => {
 	return str;
 };
 
+/**
+ * @param {Set} set the set to convert to array
+ * @return {Array} the array returned from Array.from(set)
+ */
 const getArray = set => Array.from(set);
 
+/**
+ * @param {Set<Module>} set the Set to get the count/size of
+ * @return {number} the size of the modules
+ */
 const getModulesSize = set => {
-	let count = 0;
+	let size = 0;
 	for (const module of set) {
-		count += module.size();
+		size += module.size();
 	}
-	return count;
+	return size;
 };
 
+/**
+ * @class
+ * @name Chunk
+ * @description A Chunk is a unit of encapsulation for Modules.
+ * Chunks are "rendered" into bundles that get emitted when the build completes.
+ */
 class Chunk {
+	/**
+	 * @param {string=} name of chunk being created, is optional (for subclasses)
+	 */
 	constructor(name) {
+		/** @type {number | null} */
 		this.id = null;
+		/** @type {number[] | null} */
 		this.ids = null;
+		/** @type {number} */
 		this.debugId = debugId++;
+		/** @type {string} */
 		this.name = name;
+		/** @type {Module=} */
 		this.entryModule = undefined;
+		//TODO make these typed generics for Module[] and ChunkGroup[] and their sort being (T, T): => 1,-1,0
+		//See https://github.com/webpack/webpack/pull/7046
+		/** @private */
 		this._modules = new SortableSet(undefined, sortByIdentifier);
+		/** @private */
 		this._groups = new SortableSet(undefined, sortById);
+		/** @type {Source[]} */
 		this.files = [];
+		/** @type {boolean} */
 		this.rendered = false;
+		/** @type {string=} */
 		this.hash = undefined;
+		/** @type {Object} */
 		this.contentHash = Object.create(null);
+		/** @type {string=} */
 		this.renderedHash = undefined;
+		/** @type {string=} */
 		this.chunkReason = undefined;
+		/** @type {boolean} */
 		this.extraAsync = false;
 	}
 
+	/**
+	 * @deprecated Chunk.entry has been deprecated. Please use .hasRuntime() instead
+	 * @return {never} Throws an error trying to access this property
+	 */
 	get entry() {
 		throw new Error(ERR_CHUNK_ENTRY);
 	}
 
+	/**
+	 * @deprecated .entry has been deprecated. Please use .hasRuntime() instead
+	 * @param {never} data The data that was attempting to be set
```

Comment:
this is type `any` or `*` in JSDoc syntax

Reply:
Actually never mind! I see what you're trying to do here! This is clever! Type system will complain if this setter is ever called.

Reply:
Yup exactly!